docs/pages/inboxes/user-consent/support-user-consent.md

```diff
@@ -132,38 +136,183 @@ try await conversation.updateConsent(.allowed) // .allowed | .denied
   // Received consent
-}```
+}
+```
+
+:::
+
+## Update consent for an individual in a group chat
+
+Update the consent state for an individual in a group chat:
+
+:::tip[Note]
+You may want to enable users to deny or allow a users on an individual basis. You can then update the group chat UI to hide messages from denied individuals.
+:::
+
+:::code-group
+
+```js [Browser]
+import { ConsentEntityType, ConsentState } from "@xmtp/browser-sdk";
+
+await client.setConsentStates([
+  {
+    entityId: inboxId,
+    entityType: ConsentEntityType.InboxId,
+    state: ConsentState.Denied,
+  },
+]);
+```
+
+```js [Node]
+import { ConsentEntityType, ConsentState } from "@xmtp/node-sdk";
+
+// set consent state from the client (can set multiple states at once)
+await client.setConsentStates([
+  {
+    entityId: inboxId,
+    entityType: ConsentEntityType.InboxId,
+    state: ConsentState.Denied,
+  },
+]);
+```
+
+```tsx [React Native]
+await client.preferences.setConsentState(
+  new ConsentRecord(inboxId, 'inbox_id', 'denied')
+)
+```
+
+```kotlin [Kotlin]
+client.preferences.setConsentState(
+    listOf(
+        ConsentRecord(
+            inboxId,
+            EntryType.INBOX_ID,
+            ConsentState.DENIED
+        )
+    )
+)
+```
+
+```swift [Swift]
+try await client.preferences.setConsentState(
+  entries: [
+    ConsentRecord(
+      value: inboxID, 
+      entryType: .inbox_id,
+      consentType: .denied)
+  ])
+```
 
 :::
 
+## Get the consent state of an individual in a group chat
+
+Get the consent state of an individual in a group chat:
+
+:::tip[Note]
+You may want to enable users to deny or allow a users on an individual basis. You can then update the group chat UI to hide messages from denied individuals.
+:::
+
+:::code-group
+
+```js [Browser]
+import { ConsentEntityType } from "@xmtp/browser-sdk";
+
+const inboxConsentState = await client.getConsentState(
+  ConsentEntityType.InboxId,
+  inboxId
+);
+```
+
+```js [Node]
+import { ConsentEntityType } from "@xmtp/node-sdk";
+
+const inboxConsentState = await client.getConsentState(
+  ConsentEntityType.InboxId,
+  inboxId
+);
+```
+
+```tsx [React Native]
+// Get consent directly on the member
+const memberConsentStates = (await group.members()).map(
+  (member) => member.consentState()
+)
+
+// Get consent from the inboxId
+const inboxConsentState = await client.preferences.inboxIdConsentState(inboxId)
+```
+
+```kotlin [Kotlin]
+// Get consent directly on the member
+val memberConsentStates = group.members().map { it.consentState }
+
+// Get consent from the inboxId
+val inboxConsentState = client.preferences.inboxIdState(inboxId)
+```
+
+```swift [Swift]
+// Get consent directly on the member
+let memberConsentStates = try await group.members.map(\.consentState)
+
+// Get consent from the inboxId
+let inboxConsentState = try await client.preferences.inboxIdState(inboxId: inboxId)
+```
+
+:::
+
+## See who created and added you to a group
```

Comment:
Not sure if this suggested edit is correct:

```suggestion
## See who created a group and added a user to it
```

Reply:
The method is `group.addedByInboxId` so it really only tells you who added you to the group. So you can't see who added someone else with this method.

Reply:
So as long as thats clear I'm fine with the edit.

Reply:
Ah, got it - your original conveys this detail more clearly!